src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
+
+    public ArrayList() {
+        arrayOfElements = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (actualSize == arrayOfElements.length) {
+            increaseCapacity();
+            arrayOfElements[actualSize] = value;
+            actualSize++;
+        } else {
+            arrayOfElements[actualSize] = value;
+            actualSize++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > actualSize) {
+            throw new ArrayListIndexOutOfBoundsException("Index: " + index + " is out of bounds");
+        }
+        if (actualSize == arrayOfElements.length) {
+            increaseCapacity();
+        }
+        System.arraycopy(arrayOfElements, index, arrayOfElements, index + 1, actualSize - index);
+        arrayOfElements[index] = value;
+        actualSize++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
+        if (checkIndexInRange(index)) {
+            return arrayOfElements[index];
+        }
         return null;
     }
 
     @Override
     public void set(T value, int index) {
-
+        if (checkIndexInRange(index)) {
+            arrayOfElements[index] = value;
+        }
     }
 
     @Override
     public T remove(int index) {
+        if (checkIndexInRange(index)) {
+            final T removedElement = arrayOfElements[index];
+            System.arraycopy(arrayOfElements, index + 1, arrayOfElements,
+                    index, actualSize - index - 1);
+            actualSize--;
+            return removedElement;
+        }
         return null;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int i = 0; i < actualSize; i++) {
+            if (arrayOfElements[i] == element || arrayOfElements[i] != null
+                    && arrayOfElements[i].equals(element)) {
+                return remove(i);
+            }
+        }
+        throw new NoSuchElementException("Element: " + element + " is not found");
     }
 
     @Override
     public int size() {
-        return 0;
+        return actualSize;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return actualSize == 0;
+    }
+
+    private boolean checkIndexInRange(int index) {
+        if (index < 0 || index >= actualSize) {
+            throw new ArrayListIndexOutOfBoundsException("Index: " + index + " is out of bounds");
+        }
+        return true;
+    }
+
+    private void increaseCapacity() {
+        T[] temporaryArray = (T[]) new Object[(int) (arrayOfElements.length * INCREASE_RATE)];
+        System.arraycopy(arrayOfElements, 0, temporaryArray, 0, actualSize);
+        arrayOfElements = temporaryArray;
+    }
+
+    private T removeElementAndResize(int index) {
```

Comment:
as said earlier, you are not actually changing the size of the array, so the naming is a bit off.
Also, this method is an exact duplicate of `remove(index)`, and is not used anywhere, so remove it